umh-core/pkg/config/manager.go

```diff
@@ -59,14 +84,16 @@ type FileConfigManager struct {
 	// logger is the logger for the config manager
 	logger *zap.SugaredLogger
 
-	// mutexReadAndWrite for full cycle read and write access to the config file
-	mutexReadAndWrite sync.Mutex
+	// mutexAtomicUpdate for full cycle read and write access (atomic update) to the config file
+	mutexAtomicUpdate sync.Mutex
 
 	// simple mutex for read access or write access to the config file
-	mutexReadOrWrite sync.Mutex
+	mutexReadOrWrite sync.RWMutex
```

Comment:
when to use mutexReadOrWrite:
1. it will be used by GetConfig or writeConfig
2. writeConfig is only to be used in this package through atomic functions

the goal is to:
1. allow multiple GetConfigs / read operations
2. but prevent reading and writing at the same time